### PR TITLE
fix: use scalar for torch.sinh, torch.acosh on f32 inputs (aten)

### LINKED ISSUE #183763: torch.compile overflows torch.sinh near float32 boundary while eager returns finite values
### 🐛 Describe the bug

`torch.compile` with the Inductor backend produces different overflow behavior from eager execution for `torch.sinh` on large float32 inputs near the overflow boundary.

Eager execution returns finite float32 values around `1e38`, while the compiled version returns `inf` / `-inf` for the same inputs.

```
import warnings
warnings.filterwarnings("ignore")

import numpy as np
import torch

def fn(x):
    return torch.sinh(x)

x = torch.tensor([88.85, 89.2, 88.95, -89.0], dtype=torch.float32)

eager = fn(x).cpu().numpy()

torch._dynamo.reset()
compiled = torch.compile(fn, backend="inductor", fullgraph=True)(x).cpu().numpy()

print(f"input: {x.cpu().numpy()}")
print(f"eager: {eager}")
print(f"comp : {compiled}")

assert np.isfinite(eager[0]) and eager[0] > 1e38
assert np.isinf(comp[0])
```

### Error logs

input: [ 88.85  89.2   88.95 -89.  ]
eager: [ 1.9321198e+38  2.7418043e+38  2.1353193e+38 -2.2448064e+38]
comp : [ inf  inf  inf -inf]

### Versions

PyTorch version: 2.13.0.dev20260513+cpu
Is debug build: False
CUDA used to build PyTorch: Could not collect
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.3 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04.1) 13.3.0
Clang version: 18.1.3 (1ubuntu1)
CMake version: Could not collect
Libc version: glibc-2.39

Python version: 3.11.15 (main, Mar 11 2026, 17:20:07) [GCC 14.3.0] (64-bit runtime)
Python platform: Linux-6.17.0-20-generic-x86_64-with-glibc2.39
Is CUDA available: False
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration:
GPU 0: NVIDIA RTX 6000 Ada Generation
GPU 1: NVIDIA RTX 6000 Ada Generation

Nvidia driver version: 570.211.01
cuDNN version: Could not collect
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

Versions of relevant libraries:
[pip3] numpy==2.4.4
[pip3] torch==2.13.0.dev20260513+cpu
[conda] numpy                      2.4.4                   pypi_0           pypi
[conda] torch                      2.13.0.dev20260513+cpu  pypi_0           pypi

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

### PR DESCRIPTION
**Pull request by evacchi**
closes #183763, #183765, #183768

It looks like the underlying issue is all related to Sleef handling of f32 (see https://github.com/shibatch/sleef/issues/364)

We replace Sleef's float `cosh/sinh/acosh` calls with `map(std::cosh)` etc. in all 6 vectorized wrapper files (NEON, AVX2, AVX-512, SVE, VSX, z/Arch). This falls back to the scalar C library which handles the full float32 range correctly. Double and half-precision paths are unaffected and left unchanged.

Of course, this also means that, for this code path, the underlying computation is no longer actually vectorized.

## Details

`torch.cosh`, `torch.sinh`, and `torch.acosh` return inf (or a wrong result) for float32 inputs near the overflow boundary, both in eager mode (large tensors) and under `torch.compile`.

The cause appears to be in `PyTorch's Vectorized<float>::cosh()` (and `sinh/acosh`), calling Sleef's SIMD implementations (Sleef_coshf4_u10, etc.). Sleef's `xcoshf` internally computes `exp(x)` using `expk2f`, which represents the result as a double-float pair (two floats for extra precision).

This extends precision but not range; the high word is still a float, so exp(88.73) overflows to inf even though cosh(88.73) ≈ 1.7e38 is well within float32 range (max ~3.4e38).

Sleef documents the valid input domain for sinhf as [-88.5, 88.5] -- see https://sleef.org/purec.xhtml#Sleef_sinh_u10 referenced in https://github.com/shibatch/sleef/issues/364; inputs outside this range may return inf even when the result is representable in float32.

The scalar C library std::coshf doesn't have this problem because it uses actual double (64-bit) for intermediate computation, which can represent exp(88.73) ≈ 3.4e38 without overflow.

The bug is in a single implementation (sleefsimdsp.c) that gets compiled for all SIMD targets, so all architectures are affected.

Note: a related overflow in `asinh` was recently fixed at the Inductor codegen level in #184105. The same Sleef root cause should apply there; the `Vectorized<float>::asinh()` scalar fallback could be done in this PR or as a follow-up

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @jerryzh168 @aditew01 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

**Comment by evacchi** [comment]
uhm, advice wanted for the label, this is in fact sort-of user facing -- we should probably [update the docs](https://docs.pytorch.org/docs/2.12/generated/torch.sinh.html) 

> When input is on the CPU, the implementation of torch.sinh may use the Sleef library, which rounds very large results to infinity or negative infinity. See [here](https://sleef.org/purec.xhtml) for details.

**Comment by evacchi** [comment]
just noticed the ongoing discussion at https://github.com/pytorch/pytorch/pull/183909 and   https://github.com/pytorch/pytorch/issues/183767#issuecomment-4460358301 -- happy to comply with whatever decision/close this PR :)

**Comment by jansel** [comment]
@claude review these changes

**Comment by evacchi** [comment]
addressed comments, left the BFloat16 issue for a follow-up because the change is a bit more articulate: vectorized wrappers use a map that takes a SIMD function `(__m256 -> __m256)`, not a scalar function. There's no scalar `map(float -> float)` so we can't just swap in `std::cosh`. If the maintainers believe it makes sense to address it here, we can do it though.

but make sure you check the concerns in https://github.com/pytorch/pytorch/issues/183767#issuecomment-4460358301

**Review by jgong5 — APPROVED** [review]
LGTM. But why do you change the third_party?

**Comment by evacchi** [comment]
ough inadvertently `git commit -asm` will revert

**Comment by evacchi** [comment]
fixed

**Comment by evacchi** [comment]
@jgong5 @jansel I think this should be good to go

**Review by jansel — APPROVED** [review]
